lib/sisimai/bite/email/domino.rb

```diff
@@ -0,0 +1,174 @@
+      }
+      Re1 = {
+        :begin   => %r/\AYour message/,
+        :rfc822  => %r|\AContent-Type: message/delivery-status\z|,
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        userunknown: %r{(?>
+           not[ ]listed[ ]in[ ](?:
+             Domino[ ]Directory
+            |public[ ]Name[ ][&][ ]Address[ ]Book
+            )
+          |Domino[ ]ディレクトリには見つかりません
+          )
+        }x,
+        filtered:    %r/Cannot[ ]route[ ]mail[ ]to[ ]user/x,
+        systemerror: %r/Several[ ]matches[ ]found[ ]in[ ]Domino[ ]Directory/x,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'IBM Domino Server'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from IBM Domino Server
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        subjecttxt = ''     # (String) The value of Subject:
+        v = nil
+
+        hasdivided.each do |e|
+          next if e.empty?
+
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:'message-rfc822']).zero?` instead of `readcursor & Indicators[:'message-rfc822'] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->